### PR TITLE
KAFKA-5354: MirrorMaker not preserving headers

### PR DESCRIPTION
**Pull request by michaelandrepearce**
Add test case to ensure fix and avoid regression
Update mirror maker for new consumer to preserve headers

### CODE REVIEW

File: core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala [hunk 1]
```diff
@@ -55,4 +55,20 @@ class MirrorMakerTest {
...
+    val result = MirrorMaker.defaultMirrorMakerMessageHandler.handle(consumerRecord)
+    assertEquals(1, result.size)
+
+    val producerRecord = result.get(0)
+    assertEquals(now, producerRecord.timestamp)
+    assertEquals("topic", producerRecord.topic)
+    assertNull(producerRecord.partition)
+    assertEquals("key", new String(producerRecord.key))
+    assertEquals("value", new String(producerRecord.value))
+    assertEquals("headerValue", new String(producerRecord.headers().lastHeader("headerKey").value()))
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/tools/MirrorMakerTest.scala]: It would be good to assert that there is a single header.

**Review by ijuma — APPROVED** [review]
Thanks for the PR, looks good overall, just one comment.

**Comment by michaelandrepearce** [comment]
@ijuma thanks, added

**Comment by michaelandrepearce** [comment]
Retest this please

**Comment by ijuma** [comment]
I merged #3322 to trunk and 0.11. Please close this PR.

**Pull request closed without merging**